footer.php

```diff
@@ -332,3 +334,102 @@ function make_navbutton($imgsrc, $bgcolour, $title, $url, $classes = null, $neww
     }
     return $output;
 }
+
+/**
+ * Create Activity completion buttons
+ *
+ * @return string
+ */
+function make_activitycomplete_button() {
+
+    global $COURSE, $DB, $CFG, $PAGE, $OUTPUT;
+
+    $output = "";
+    $style = "";
+
+    if (!$settings = $DB->get_record('navbuttons', array('course' => $COURSE->id))) {
```

Comment:
We already have $settings at the point where this is called - let's save a DB query by passing it in as a function param to make_activitycomplete_button()